src/analysis/vector/geometry_checker/qgsgeometrycurvecheck.cpp

```diff
@@ -0,0 +1,85 @@
+/***************************************************************************
+    qgsgeometrycurvecheck.cpp
+    ---------------------
+    begin                : December 2023
+    copyright            : (C) 2023 by Loïc Bartoletti
+    email                : loic dot bartoletti at oslandia dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsabstractgeometry.h"
+#include "qgsgeometrycurvecheck.h"
+#include "qgsfeaturepool.h"
+#include "qgsgeometrycheckerror.h"
+
+void QgsGeometryCurveCheck::collectErrors( const QMap<QString, QgsFeaturePool *> &featurePools, QList<QgsGeometryCheckError *> &errors, QStringList &messages, QgsFeedback *feedback, const LayerFeatureIds &ids ) const
+{
+  Q_UNUSED( messages )
+  const QMap<QString, QgsFeatureIds> featureIds = ids.isEmpty() ? allLayerFeatureIds( featurePools ) : ids.toMap();
+  const QgsGeometryCheckerUtils::LayerFeatures layerFeatures( featurePools, featureIds, compatibleGeometryTypes(), feedback, mContext );
+  for ( const QgsGeometryCheckerUtils::LayerFeature &layerFeature : layerFeatures )
+  {
+    const QgsAbstractGeometry *geom = layerFeature.geometry().constGet();
+    for ( int iPart = 0, nParts = geom->partCount(); iPart < nParts; ++iPart )
+    {
+      const QgsAbstractGeometry *part = QgsGeometryCheckerUtils::getGeomPart( geom, iPart );
+      if ( part->hasCurvedSegments() )
+      {
+        errors.append( new QgsGeometryCheckError( this, layerFeature, part->centroid(), QgsVertexId( iPart ) ) );
+      }
+    }
+  }
+}
+
+void QgsGeometryCurveCheck::fixError( const QMap<QString, QgsFeaturePool *> &featurePools, QgsGeometryCheckError *error, int method, const QMap<QString, int> & /*mergeAttributeIndices*/, Changes &changes ) const
```

Comment:
`int method` is an enum, why are you using an `int`?

Reply:
Old code. Herited from https://github.com/qgis/QGIS/blob/b2685bf0e96e9367c2e486488ad9319a3324848d/src/analysis/vector/geometry_checker/qgsgeometrycheck.h#L285